### ISSUE TITLE
NinjaLoot - bug of party (BB #542)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#542]
This issue was migrated from bitbucket.
**Original Reporter:** smoldar
**Original Date:** 09.03.2014 09:44:01 GMT+0000
**Original Priority:** minor
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/542

<hr>

If someone leave from party after kill of boss. He dont need be connected with loot and can  steal all items from boss. But loot is connected with first damager, you dont need to add 1. damage to ninjaloot of boss.

Loot is connected with all players in group, i think it must be connected with 1. damager or group. After leave of group player must be blocked from loot i think.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 09.03.2014 09:51:26 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/542/#comment-9026913

<hr>

I tested, MasterLoot is ok, but this problem is on Group Loot, after kill can leave and loot all items without raid, if you are faster in loot, all will be yours.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 22.03.2014 13:52:00 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/542/#comment-9260867

<hr>

What is the expected behavior ?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 22.03.2014 14:04:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/542/#comment-9260941

<hr>

Loot must be connected with party/raid or 1. damager i think

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 22.03.2014 14:09:03 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/542/#comment-9260964

<hr>

Connected with group GUID I think.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 22.03.2014 14:34:29 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/542/#comment-9261139

<hr>

But now you can leave from raid and steal all drops. You dont need be 1. damager or leader of party for steal.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 22.03.2014 16:18:06 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/542/#comment-9261895

<hr>

I think I got it. But have to go right now, maybe you could test it ?

```
# HG changeset patch
# User Zaffy <zaffy@lunar.sk>
# Date 1395505000 -3600
#      Sat Mar 22 17:16:40 2014 +0100
# Node ID d93dd3e67c175c794af30461a24adc840c1a8404
# Parent  ed97957399d8cc8479b38938d34ac889ff851c0d
Fixed issue #542 - ninja loot

diff -r ed97957399d8 -r d93dd3e67c17 src/game/LootHandler.cpp
--- a/src/game/LootHandler.cpp  Sat Mar 22 15:13:05 2014 +0100
+++ b/src/game/LootHandler.cpp  Sat Mar 22 17:16:40 2014 +0100
@@ -82,7 +82,7 @@

         bool ok_loot = pCreature && pCreature->isAlive() == (player->getClass() == CLASS_ROGUE && pCreature->lootForPickPocketed);

-        if (!ok_loot || !pCreature->IsWithinDistInMap(_player,INTERACTION_DISTANCE))
+        if (!ok_loot || !player->isAllowedToLoot(pCreature) || !pCreature->IsWithinDistInMap(_player,INTERACTION_DISTANCE))
         {
             player->SendLootRelease(lguid);
             return;
@@ -198,7 +198,7 @@

             bool ok_loot = pCreature && pCreature->isAlive() == (player->getClass() == CLASS_ROGUE && pCreature->lootForPickPocketed);

-            if (ok_loot && pCreature->IsWithinDistInMap(_player,INTERACTION_DISTANCE))
+            if (ok_loot && player->isAllowedToLoot(pCreature) && pCreature->IsWithinDistInMap(_player,INTERACTION_DISTANCE))
                 pLoot = &pCreature->loot ;

             break;
diff -r ed97957399d8 -r d93dd3e67c17 src/game/Player.cpp
--- a/src/game/Player.cpp   Sat Mar 22 15:13:05 2014 +0100
+++ b/src/game/Player.cpp   Sat Mar 22 17:16:40 2014 +0100
@@ -7849,7 +7849,7 @@
         Creature *creature = GetMap()->GetCreature(guid);

         // must be in range and creature must be alive for pickpocket and must be dead for another loot
-        if (!creature || creature->isAlive() != (loot_type == LOOT_PICKPOCKETING) || !creature->IsWithinDistInMap(this,INTERACTION_DISTANCE))
+        if (!creature || !isAllowedToLoot(creature) || creature->isAlive() != (loot_type == LOOT_PICKPOCKETING) || !creature->IsWithinDistInMap(this,INTERACTION_DISTANCE))
         {
             SendLootRelease(guid);
             return;
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 23.03.2014 12:20:40 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/542/#comment-9269567

<hr>

Solves it. Good work.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 23.03.2014 12:49:25 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/542/#comment-9269786

<hr>

Fixed issue #542 - ninja loot

→ <<cset d93dd3e67c17>>
